### ISSUE TITLE
Secret environment variables doesn't handle default project id if not provided

### ISSUE THREAD

**Issue by Resousse** [google-github-actions/deploy-cloud-functions#294]
### TL;DR

As per the readme : For Secret variables  `If the project is omitted, it will be inferred from the Cloud Function project ID. ` I understand that if project ID is not provided, it should be "calculated" based on `google-github-actions/auth` project id, however, based on my tests + the looking of the code, it's not. 

### Expected behavior

`secret_environment_variables: "KEY_MAIL=KEY_MAIL"` 
should map to `KEY_MAIL`, the `latest` version of the `KEY_MAIL` secret of the `default` project ID

### Observed behavior

`unknown format. Secrets should be of the format "projects/p/secrets/s/versions/v"`

### Action YAML

```yaml
`secret_environment_variables: "KEY_MAIL=KEY_MAIL"`
```

**Comment by sethvargo** [comment]
Hi @Resousse - thank you for opening a bug and a PR. I'm not a huge fan of gcloud's "magic" here, since it isn't declarative or portable (changing a project ID has drastic effects). I personally think deploy-cloud-functions is doing the correct thing here by requiring a full resource ID, but I'd like to hear what other folks think @averikitsch @bharathkkb.

**Comment by Resousse** [comment]
Hello @sethvargo , I understand your point!
I currently use a private github project to deploy my two GCP environments. I'm with free tier, so I don't have access to `environments` which could help me a lot to manage different set of env variables values depending of the env (prod vs non prod).
So I'm currently dealing with crazy `if` logic like ` if [[ "${{github.base_ref}}" == "master" || "${{github.base_ref}}" == "main" || "${{github.ref}}" == "refs/heads/master" || "${{github.ref}}" == "refs/heads/main" ]]; then` to define if I'm in prod/main or test environment to load the good variable values :)

This feature/bug could at least help me, even if it's not the best solution, I agree.

**Comment by sethvargo** [comment]
Hi @Resousse thank you for the reply. Could you explain your use case more or add some sample YAML? I'm not sure I completely understand the problem.

**Comment by Resousse** [comment]
@sethvargo , ok, I have two environments, with 2 distincts secrets env variables. I would like to have a common github actions workflow that can work either for production or testing.
So I wanted to use the `secret_environment_variables` parameters with only a secret name, and not a version or a project id.

Then I realized the documentation was not totally aligned with what the code does, and then I found the opportunity to contribute to improve this, with my PR.

On my side I found a way to include the project id dynamically, to have a working workflow for any environment, so my personal issue is closed.

But then, it still remains the issue of either documentation or code, depending of if you want to have this kind of behavior or not (If not, documentation should be changed a bit to remove the ambiguous sentence).

**Comment by sethvargo** [comment]
Okay - I think the readme is the incorrect statement here. I'll submit a PR to fix that.
